packages/core/src/components/layout/layout.ts

```diff
@@ -0,0 +1,213 @@
+				.attr("x", (d: any) => d.x0)
+				.attr("y", (d: any) => d.y0);
+
+		enteringSVGs.merge(svg.selectAll(`svg.layout-child-${this._instanceCount}`))
+			.each(function(d: any) {
+				// Set parent component for each child
+				d.data.components.forEach(itemComponent => {
+					itemComponent.setParent(select(this));
+
+					// Render preffered & fixed items
+					const growth = Tools.getProperty(d, "data", "growth", "x");
+					if (growth === LayoutGrowth.PREFERRED || growth === LayoutGrowth.FIXED) {
+						itemComponent.render(animate);
+					}
+				});
+			});
+
+		svg.selectAll(`svg.layout-child-${this._instanceCount}`)
+		.each(function(d: any) {
+			// Calculate preffered children sizes after internal rendering
+			const growth = Tools.getProperty(d, "data", "growth", "x");
+			const matchingSVGDimensions = DOMUtils.getSVGElementSize(select(this), { useBBox: true });
+			if (growth === LayoutGrowth.PREFERRED) {
+				const matchingSVGWidth = horizontal ? matchingSVGDimensions.width : matchingSVGDimensions.height;
+				const svgWidth = horizontal ? width : height;
+
+				d.data.size = (matchingSVGWidth / svgWidth) * 100;
+			}
+		});
+
+		updatedSVGs
+			.exit()
+			.remove();
+
+		// Run through stretch x-items
+		this.children
+			.filter(child => {
+				const growth = Tools.getProperty(child, "growth", "x");
+				return growth === LayoutGrowth.STRETCH;
+			})
+			.forEach((child, i) => {
+				child.size = (100 - (+this.getPrefferedAndFixedSizeSum())) / (+this.getNumOfStretchChildren());
+			});
+
+		setTimeout(() => {
+			// Pass children data to the hierarchy layout
+			// And calculate sum of sizes
+			root = hierarchy({
+				children: hierarchyChildren
+			})
+			.sum((d: any) => d.size);
+
+			// Compute the position of all elements within the layout
+			treemap()
+				.tile(tileType)
+				.size([width, height])
+				.padding(0)
+				(root);
+
+			// Add new SVGs to the DOM for each layout child
+			svg
+				.selectAll(`svg.layout-child-${this._instanceCount}`)
+				.data(root.leaves(), (d: any) => d.data.id)
+				.attr("x", (d: any) => d.x0)
+				.attr("y", (d: any) => d.y0)
+				.attr("width", (d: any) => d.x1 - d.x0)
+				.attr("height", (d: any) => d.y1 - d.y0)
+				.each(function(d: any, i) {
+					d.data.components.forEach(itemComponent => {
+						const growth = Tools.getProperty(d, "data", "growth", "x");
+						if (growth === LayoutGrowth.STRETCH) {
+							itemComponent.render(animate);
+						}
+					});
+				});
+		}, 0);
+	}
+
+	// Pass on model to children as well
+	setModel(newObj) {
+		super.setModel(newObj);
+
+		this.children.forEach(child => {
+			child.components.forEach(component => component.setModel(newObj));
+		});
+	}
+
+	// Pass on essentials to children as well
+	setServices(newObj) {
+		super.setServices(newObj);
+
+		this.children.forEach(child => {
+			child.components.map(component => component.setServices(newObj));
```

Comment:
Probably should be a `forEach` unless we're doing something with the value

Reply:
good catch